File: ruby/equalwords.rb
text=File.open("words.txt", "r:UTF-8").read
collection = Hash.new{|hsh,key| hsh[key] = [] }

text.each_line do |word|
  abc_word = word.split("").sort.join("")
  collection[abc_word].push word.strip
end

collection.each_key do |k|
  print "#{collection[k].join(", ")}\n"
end

File: javascript/equalwords.js
var fs = require('fs');
var dict = {};
var array = fs.readFileSync('../words.txt','utf8').split("\n");

for (let i in array) {
  var abc_word = array[i].split('').sort().join('');
  if (abc_word in dict) {
    dict[abc_word].push(array[i].trim());
  } else {
    dict[abc_word] = [array[i].trim()];
  }
}

for (let i in dict) {
  console.log(dict[i].join(', '))
}
